wpas-hl7-translator/src/main/java/wales/nhs/dhcw/inthub/wpasHl7/MprToAdtA40Mapper.java

```diff
@@ -0,0 +1,172 @@
+            adtMessage.getPATIENT().getPID().getPid13_PhoneNumberHome(2).getXtn9_AnyText().setValue("Mobile");
+
+            adtMessage.getPATIENT().getPID().getPid16_MaritalStatus().getCe1_Identifier().setValue(transaction.getMARITALSTATUS());
+            adtMessage.getPATIENT().getPID().getPid17_Religion().getCe1_Identifier().setValue(transaction.getRELIGIONSTATUS());
+            adtMessage.getPATIENT().getPID().getPid22_EthnicGroup(0).getCe1_Identifier().setValue(transaction.getETHNICORIGIN());
+            if (transaction.getDEATHDATE() != null && !transaction.getDEATHDATE().isBlank()) {
+                LocalDate date1 = LocalDate.parse(transaction.getBIRTHDATE(), DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                String deathDate = date1.format(inputdateFormatter);
+                adtMessage.getPATIENT().getPID().getPid29_PatientDeathDateAndTime().getTs1_Time().setValue(deathDate);
+                adtMessage.getPATIENT().getPID().getPid30_PatientDeathIndicator().setValue("Y");
+            } else {
+                adtMessage.getPATIENT().getPID().getPid29_PatientDeathDateAndTime().getTs1_Time().setValue(transaction.getDEATHDATE()); //yyyyMMdd
+                adtMessage.getPATIENT().getPID().getPid30_PatientDeathIndicator().setValue("N");
+            }
+            adtMessage.getPATIENT().getPID().getPid32_IdentityReliabilityCode(0).setValue(transaction.getNHSCERTIFICATION());
+
+            DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+            LocalDateTime dateTime = LocalDateTime.parse(transaction.getUPDATEDATE(), inputFormatter);
+            adtMessage.getPATIENT().getPID().getPid33_LastUpdateDateTime().getTs1_Time().setValue(dateTime.format(requiredDateTimeFormatter));
+
+            return adtMessage;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public ADT_A39 PD1Mapper(MAINDATA.TRANSACTION transaction) throws DataTypeException {
+        try {
+            adtMessage.getPATIENT().getPD1().getPd13_PatientPrimaryFacility(0).getXon1_OrganizationName().setValue(transaction.getGPADDR1());
+            adtMessage.getPATIENT().getPD1().getPd13_PatientPrimaryFacility(0).getXon10_OrganizationIdentifier().setValue(transaction.getGPPRACTICE());
+            adtMessage.getPATIENT().getPD1().getPd14_PatientPrimaryCareProviderNameIDNo(0).getXcn1_IDNumber().setValue(transaction.getREGISTEREDGP());
+            adtMessage.getPATIENT().getPD1().getPatientPrimaryCareProviderNameIDNo(0).getXcn2_FamilyName().getFn1_Surname().setValue(transaction.getGPSURNAME());
+            adtMessage.getPATIENT().getPD1().getPatientPrimaryCareProviderNameIDNo(0).getXcn3_GivenName().setValue(transaction.getGPINITS());
+
+            return adtMessage;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
please don't map all the Exceptions to runtime 
 this method defines throwing DataTypeException, add aditional types of exceptions here